src/main/java/com/vg/web/db/BaseJsonRedisDao.java

```diff
@@ -266,60 +270,55 @@ protected void delete(Transaction tx, String id) {
         tx.zrem(kMtime, id);
     }
 
-    private final Multimap<String, PubSubUpdateListener> listeners = ArrayListMultimap.create();
-    private final List<PubSubUpdateListener> allMessagesListeners = new CopyOnWriteArrayList<>();
+    private final Map<String, Subscription> listeners = new ConcurrentHashMap<>();
 
-    PubSubUpdateListener mainListener = id -> {
-        synchronized (listeners) {
-            if (listeners.containsKey(id)) {
-                listeners.get(id).forEach(x -> x.accept(id));
-            }
-            allMessagesListeners.forEach(x -> x.accept(id));
-        }
-    };
     private Field revisionField;
-    private Subscription subscription;
 
+    @Deprecated
```

Comment:
@chugai breaks a lot of code and it is not compatible with version `1.6.5` in version `1.7.0` we can delete it